index.html

```diff
@@ -208,6 +209,30 @@ <h3>GetUserMediaSemantics enum</h3>
         </table>
       </div>
     </section>
+    <section id="mediatrackconstraintset-dictionary-extensions">
+      <h3>MediaTrackConstraintSet dictionary extensions</h3>
+      <div>
+        <pre class="idl"
+>partial dictionary MediaTrackConstraintSet {
```

Comment:
Done: I added the constraint to MediaTrackSupportedConstraints, MediaTrackCapabilities and MediaTrackSettings (in addition to MediaTrackConstraintSets).

> getUserMedia should reject a required avoidInefficientCompressedPixelFormat constraint.

Is this the same as adding it to "allowed required constraints for device selection"? I updated the PR to do this, but see other comment.

> It might also be good to write WPT tests as well.

This feature is testable with mocks controlling device capabilities or with real web cams, but AFAIK there's no way to mock device capabilities inside of WPTs? So I think it's currently not possible to test the device selection part unless WPTs are more powerful now than they used to be.

It's possible to WPT test the presence of the constraint though, which would naturally be added when implementing this.